com.unity.render-pipelines.high-definition/Editor/VFXGraph/Shaders/VFXDecal.template

```diff
@@ -0,0 +1,191 @@
+
+//TODO : Check for unnecessary includes / and for necessary ones
+#define VFXComputePixelOutputToDBuffer(i, normalWS,outDBuffer) \
+{ \
+	DecalSurfaceData surfaceData; \
+    ZERO_INITIALIZE(DecalSurfaceData, surfaceData); \
+    VFXGetSurfaceDecalData(surfaceData,i,normalWS); \
+ \
+    ENCODE_INTO_DBUFFER(surfaceData, outDBuffer); \
+}
+
+void VFXGetSurfaceDecalData(out DecalSurfaceData surfaceData, VFX_VARYING_PS_INPUTS i, float3 normalWS)
+{
+	ZERO_INITIALIZE(DecalSurfaceData, surfaceData);
+	VFXTransformPSInputs(i);
+
+${VFXLoadParameter:{enableDecalLayers}}
+${VFXLoadParameter:{decalLayerMask}}
+
+DecalPrepassData material;
+ZERO_INITIALIZE(DecalPrepassData, material);
+//Check light layer
+if(_EnableDecalLayers)
+{
+	// Clip the decal if it does not pass the decal layer mask of the receiving material.
+	// Decal layer of the decal
+	DecodeFromDecalPrepass(i.pos.xy, material);
+
+	if ((decalLayerMask & material.decalLayerMask) == 0)
```

Comment:
note: haven't think more about this, but I notice a lot of duplicate code with the regular shader code and I am wondering how we could share more of it. + In the future we will certainly add decal sahder graph support in VFX so we need to think about how to reduce code duplcaition.

Reply:
To be honest, I mostly followed the structure from the lit outputs of the VFX Graph. But I agree, there is a lot of duplicate code, which won't help with maintainability.
I'm not familiar with the new SG integration, I indeed need to assess if it will make it more diffcult.

Reply:
Yes idea is at some point to rely completely on SG for render pipe specific code and simplify/factorize our templates a lot. But we're not there yet